internal/querycoord/global_meta_broker.go

```diff
@@ -502,3 +504,55 @@ func (broker *globalMetaBroker) getSegmentStates(ctx context.Context, segmentID
 
 	return resp.States[0], nil
 }
+
+func (broker *globalMetaBroker) acquireSegmentsReferLock(ctx context.Context, segmentIDs []UniqueID) error {
+	ctx, cancel := context.WithTimeout(ctx, timeoutForRPC)
+	defer cancel()
+	acquireSegLockReq := &datapb.AcquireSegmentLockRequest{
+		SegmentIDs: segmentIDs,
+		NodeID:     Params.QueryCoordCfg.GetNodeID(),
+	}
+	status, err := broker.dataCoord.AcquireSegmentLock(ctx, acquireSegLockReq)
+	if err != nil {
+		log.Error("QueryCoord acquire the segment reference lock error", zap.Int64s("segIDs", segmentIDs),
+			zap.Error(err))
+		return err
+	}
+	if status.ErrorCode != commonpb.ErrorCode_Success {
+		log.Error("QueryCoord acquire the segment reference lock error", zap.Int64s("segIDs", segmentIDs),
+			zap.String("failed reason", status.Reason))
+		return fmt.Errorf(status.Reason)
+	}
+
+	return nil
+}
+
+func (broker *globalMetaBroker) releaseSegmentReferLock(ctx context.Context, segmentIDs []UniqueID) error {
+	ctx, cancel := context.WithTimeout(ctx, timeoutForRPC)
+	defer cancel()
+
+	releaseSegReferLockReq := &datapb.ReleaseSegmentLockRequest{
+		SegmentIDs: segmentIDs,
+		NodeID:     Params.QueryCoordCfg.GetNodeID(),
+	}
+
+	if err := retry.Do(ctx, func() error {
+		status, err := broker.dataCoord.ReleaseSegmentLock(ctx, releaseSegReferLockReq)
+		if err != nil {
+			log.Error("QueryCoord release reference lock on segments failed", zap.Int64s("segmentIDs", segmentIDs),
+				zap.Error(err))
+			return err
+		}
+
+		if status.ErrorCode != commonpb.ErrorCode_Success {
+			log.Error("QueryCoord release reference lock on segments failed", zap.Int64s("segmentIDs", segmentIDs),
+				zap.String("failed reason", status.Reason))
+			return errors.New(status.Reason)
+		}
+		return nil
+	}, retry.Attempts(100)); err != nil {
```

Comment:
why we panic but not throw error?

Reply:
No, the error in postExecute will not be caught, and if it fails to release the lock, it should panic, so that datacoord can release the lock. 
Otherwise the lock will not be released.